aleksei-bulgak-study/nodejs-aws-be
# Task 9 (Backend For Frontend)

Provide your reviewers with the link to the repo, product and CART services URLs and bff-service URL
bff endpoint=http://aleksei-bulgak-study-bff-api.eu-west-1.elasticbeanstalk.com
cart=https://q6alx46vvh.execute-api.eu-west-1.amazonaws.com/prod/api
products=https://kutdlurimk.execute-api.eu-west-1.amazonaws.com/dev/products

Example of endpoints:
http://aleksei-bulgak-study-bff-api.eu-west-1.elasticbeanstalk.com/products
http://aleksei-bulgak-study-bff-api.eu-west-1.elasticbeanstalk.com/cart/profile/cart

Cache was added only for endpoints which name starts from `products` and ttl configured for 120 seconds=2mins

Evaluation criteria:
    How to test:
        Get products list
        Create new product
        Get products list - result shouldn’t have new product
        Wait more than 2 minutes
        Get products list - result should have new product
* [x] 1 - Use NestJS to create bff-service instead of express (NestJs application implemented in separate folder https://github.com/aleksei-bulgak-study/nodejs-aws-be/tree/task9/bff-service-nest)


Description:
.env file contains only 2 lines:
cart=https://q6alx46vvh.execute-api.eu-west-1.amazonaws.com/prod/api
products=https://kutdlurimk.execute-api.eu-west-1.amazonaws.com/dev/products

To create product you can use this file https://github.com/aleksei-bulgak-study/nodejs-aws-be/blob/task9/importService/data/test.csv
And upload it via FE application https://d38r3x11fba1qn.cloudfront.net
